src/utils.rs

```diff
@@ -445,9 +445,27 @@ pub fn format_url(url: &str) -> String {
 }
 
 // Rewrite Reddit links to Libreddit in body of text
-pub fn rewrite_urls(text: &str) -> String {
-	match Regex::new(r#"href="(https|http|)://(www.|old.|np.|amp.|)(reddit).(com)/"#) {
-		Ok(re) => re.replace_all(text, r#"href="/"#).to_string(),
+pub fn rewrite_urls(input_text: &str) -> String {
+	let text1 = match Regex::new(r#"href="(https|http|)://(www.|old.|np.|amp.|)(reddit).(com)/"#) {
+		Ok(re) => re.replace_all(input_text, r#"href="/"#).to_string(),
+		Err(_) => String::new(),
+	};
+
+	// Remove links to Giphy
+	let text2 = match Regex::new(r#"<a href=".*giphy\.com.*".*[A-z]">(?P<img>.*)</a>"#) {
+		Ok(re) => re.replace(&text1, "$img").to_string(),
+		Err(_) => String::new(),
+	};
+
+	// Rewrite external media previews to Libreddit
+	match Regex::new(r"https://external-preview\.redd\.it(.*)[^?]") {
+		Ok(re) => {
+			if re.is_match(&text2) {
+				re.replace_all(&text2, format_url(re.find(&text2).unwrap().as_str())).to_string()
```

Comment:
[re.find()](https://docs.rs/regex/1.5.4/regex/struct.Regex.html#method.find) appears to grab the leftmost-first match. If a user was to embed multiple GIFs into their comment, wouldn't they all be proxied as the first GIF?

Reply:
Reddit does not allow adding more than 1 GIF
![Screenshot from 2021-05-08 21-06-13](https://user-images.githubusercontent.com/49381066/117557490-c72f9500-b062-11eb-8e69-a8a34e32ea8a.png)